app/components/solr.js

```diff
@@ -3,87 +3,83 @@ const { spawn } = require('child_process');
-
-        this.port = await findFreePort(8983);
-        this.zooPort = this.port + 1000;
-        log.info(`Founded free port to use for solr ${this.port}`);
-
-        let solrHome = path.join(app.getPath('home'), ".dbvtk", "index");
-        let solrPidFile = path.join(solrHome, "solr-" + this.port + ".pid");
+        if (fs.existsSync(this.solrPortFile)) {
+            // if there's a port file, use that port
+            this.port = parseInt(fs.readFileSync(this.solrPortFile, 'utf8'));
+            log.info(`Starting Solr with port: ${this.port}`);
+        } else {
+            this.port = await findFreePort(8983);
+            log.info(`Found free port to use for Solr: ${this.port}`);
+        }
 
         if (process.env.SNAP_USER_COMMON) {
             log.info("SNAP_USER_COMMON: " + process.env.SNAP_USER_COMMON);
-            solrHome = process.env.SNAP_USER_COMMON + "/.dbvtk/index";
+            this.solrHome = process.env.SNAP_USER_COMMON + CONSTANTS.DBVKT_DIRECTORY + "/" + CONSTANTS.INDEX_DIRECTORY;
+            this.processInfoDir = process.env.SNAP_USER_COMMON + CONSTANTS.DBVKT_DIRECTORY + "/" + CONSTANTS.PROCESS_INFO_DIRECTORY;
         }
 
-        if (!fs.existsSync(solrHome)) {
-            const dir = fs.mkdirSync(solrHome, { recursive: true });
+        if (!fs.existsSync(this.solrHome)) {
+            const dir = fs.mkdirSync(this.solrHome, { recursive: true });
             if (dir === undefined) {
-                throw new Error(`Failed to create Solr home directory at ${solrHome}`);
+                throw new Error(`Failed to create Solr home directory at ${this.solrHome}`);
             }
         }
 
         let solrArgs = [
             "start",
             "-c",
             "-p", this.port,
-            "-s", solrHome
+            "-s", this.solrHome
         ];
 
-        console.log(solrArgs)
-
-        this.process = spawn(solrBinPath, solrArgs, {
-            cwd: app.getAppPath().replace('app.asar', 'app.asar.unpacked') + '/',
-            env: {
-                ...process.env,
-                JAVA_HOME: java.home,
-                SOLR_PID_DIR: solrHome,
-            }
-        });
-
-        this.process.on('error', (code, signal) => {
-            throw new Error('Failed to start Solr process:', err);
-        });
-
         // Waiting for solr to start
-        log.info('Wait until ' + solrPidFile + ' exists...');
-        await waitOn({ resources: [solrPidFile] });
+        log.info('Waiting for Solr process to start...');
+        this.process = spawnSolrProcess(solrArgs);
 
-        this.solrPID = parseInt(fs.readFileSync(solrPidFile));
-        log.info('Solr PID: ' + this.solrPID);
+        fs.writeFileSync(this.solrPortFile, this.port.toString(), 'utf8');
 
         let solrURL = `${this.appUrl}:${this.port}`;
-
-        log.info("Solr at " + solrURL);
         await waitOn({ resources: [solrURL] });
```

Comment:
I don't think this is enough, try using the Solr binary with the status command to check if Solr is ready to use, or some status api.